frontend/src/components/signup.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import ReactDOM from 'react-dom/client';
+import { useState } from "react"; 
+import axios from 'axios';
+import { fchmod } from 'fs';
+import '../components/signup.css'
+
+export const SignUp = () => {
+
+    const [firstName, setFirstName] = useState<string>('');
+    const [lastName, setLastName] = useState<string>('');
+    const [email, setEmail] = useState<string>('');
+    const [password, setPassword] = useState<string>('');
+    const [confirmPassword, setConfirmPassword] = useState<string>('');
+    const [warning, setWarning] = useState<string>(''); 
+
+    const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+
+        e.preventDefault(); 
+
+            if(password === confirmPassword) {
+                setWarning('');
+                await axios.post('http://localhost:8000/signup', {
+                    firstName,
+                    lastName,
+                    email,
+                    password, 
+                })
+        
+                .then((res) => {
+                    console.log(res.data)
+                })
+        
+                .catch((err) => {
+                    setWarning('Please use a different email.');
+                })
```

Comment:
technically, we don't know that a different email is the error -- there could be issues elsewhere. We should be checking the err object to see if there is data that helps us understand what we might want to. I.e. if we have a email collision then maybe this is the right error message.

Maybe there's a bug and somehow a password of length 0 was submitted -- then we might need to surface a different error.

Maybe the login server is down, then we might need to surface something else.

No change needed in this PR but i'd like us to start thinking about what _could_ go wrong.